### ISSUE TITLE
False-positive `Attempting to enable legacy extension VK_EXT_device_fault`

### ISSUE THREAD

**Issue by corporateshark** [KhronosGroup/Vulkan-ValidationLayers#12373]
**Environment:**
 - OS: Windows 10
 - GPU and driver version: RTX 3070 Ti, 610.70
 - SDK or header version if building from repo: 1.4.350
 - Options enabled (synchronization, best practices, etc.): legacy validation

**Describe the Issue**

This Vulkan driver does not expose `VK_KHR_device_fault` but supports `VK_EXT_device_fault`. I'm initializing the only supported extension and still getting this validation error:
```
Validation Warning: [ WARNING-legacy-extension ] | MessageID = 0x27d3f166
vkCreateDevice(): Attempting to enable legacy extension VK_EXT_device_fault,
but this extension has been promoted to VK_KHR_device_fault.
```

**Expected behavior**

No validation error.

**Comment by spencer-lunarg** [comment]
So this is really a "what do we expect the legacy layer to do" 

There is half the people who want it to be "only tell me on my system what is legacy"

and some people want this as a means to be warned that while they have a solution, there is another solution

Like at this point, I might just make a setting, as seems there are people who want both

**Comment by spencer-lunarg** [comment]
(Going to get some internal WG agreement what we expect this to be... thanks @corporateshark for raising this!)

**Comment by corporateshark** [comment]
> Like at this point, I might just make a setting, as seems there are people who want both

A new extra setting for this (enable all vs enable what's available) might be helpful to reduce noise!

**Comment by spencer-lunarg** [comment]
so we discussed in https://gitlab.khronos.org/vulkan/vulkan/-/issues/4876

basically we plan to support both options, the default will be "warn me if there is a KHR in the header, even if not on my device"

A good comparison made was when you update your GCC/Clang that now supports C++23, you want a warning that even though you are using C++17, a feature you are using is deprecated... you have an option to `-Wno` that warning, but also gives you a bigger heads up picture what is going on with the language... we want this layer to also be here to warn people of the many (MANY) extension that are released

**Comment by spencer-lunarg** [comment]
Also maybe better put

> the argument was that an application developer can opt out of warnings for things they know they want to ignore, but opting in to warnings requires them to already know those things exist, which would mean they didn't need the extension to tell them about them in the first place.

**Comment by corporateshark** [comment]
@spencer-lunarg That's just what the doctor ordered! Thank you!

**Comment by spencer-lunarg** [comment]
For closure, we just merged (and will be in the next SDK at the end of the month) a new `legacy_detection_mode`

https://github.com/KhronosGroup/Vulkan-ValidationLayers/blob/main/docs/legacy_detection.md#why-is-it-reporting-extensions-my-device-doesnt-even-support

this way you can pick your level legacy detection warning, but by default, we will be loud and alert people on everything

**Comment by spencer-lunarg** [comment]
(Also open to feedback on the design `legacy_detection_mode`, just open a new issue)

**Comment by corporateshark** [comment]
@spencer-lunarg Is it possible to switch it programmatically via `VK_EXT_layer_settings`?

**Comment by spencer-lunarg** [comment]
yes! (for all settings)

example

```c++
const VkBool32 kVkTrue = VK_TRUE;
const char* mode = "ONLY_ENABLED";
VkLayerSettingEXT layer_setting[2] = {
    {"VK_LAYER_KHRONOS_validation", "legacy_detection", VK_LAYER_SETTING_TYPE_BOOL32_EXT, 1, &kVkTrue},
    {"VK_LAYER_KHRONOS_validation", "legacy_detection_mode", VK_LAYER_SETTING_TYPE_STRING_EXT, 1, &mode}
};
static VkLayerSettingsCreateInfoEXT layer_setting_ci = {VK_STRUCTURE_TYPE_LAYER_SETTINGS_CREATE_INFO_EXT, nullptr, 2, layer_setting};
```
